### PR TITLE
Move RecordingData related classes to internal-api module

### PR DESCRIPTION
**Pull request by jbachorik**
# What Does This Do
Refactors a bunch of the profiler API classes and moves them to internal-api package.

# Motivation
A preparation step for graalvm native image profiling support

# Additional Notes

Jira ticket: [PROF-8562]

[PROF-8562]: https://datadoghq.atlassian.net/browse/PROF-8562?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by richardstartin — APPROVED** [review]

File: dd-java-agent/agent-jmxfetch/integrations-core
Comment (ygree) [inline dd-java-agent/agent-jmxfetch/integrations-core:1]: Was this submodule update intentional?

Comment (jbachorik) [reply dd-java-agent/agent-jmxfetch/integrations-core:1]: 😱 No ... I swear I didn't touch that module. I will try to revert this if I figure out what actually triggered the update ...

Comment (jbachorik) [reply dd-java-agent/agent-jmxfetch/integrations-core:1]: Ok, I have updated the commit not to include the changes to integrations-core.

**Review by ygree — APPROVED** [review]
LGTM

**Pull request merged**